core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -4528,16 +4543,33 @@ private void processO3Block(
                             .$(", memUsed=").$(Unsafe.getMemUsed())
                             .I$();
 
-                    if (partitionTimestamp < lastPartitionTimestamp) {
-                        // increment fixedRowCount by number of rows old partition incremented
-                        this.txWriter.fixedRowCount += partitionSize - srcDataMax;
-                    } else if (partitionTimestamp == lastPartitionTimestamp) {
-                        // this is existing "last" partition, we can set the size directly
-                        prevTransientRowCount = partitionSize;
+                    if (!partitionIsReadOnly) {
```

Comment:
it would read better if I changed the check to have the least common branch (partition is read only) upfront and then I could be able to remove one indentation level. However in the past it has been pointed out to me that the most common branch needs to be at the top, hence the "confusing" code.